presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -551,7 +556,12 @@ else if (redistributeWrites) {
                 }
             }
 
-            if (partitioningScheme.isPresent() && !source.getProperties().isCompatibleTablePartitioningWith(partitioningScheme.get().getPartitioning(), false, metadata, session)) {
+            if (partitioningScheme.isPresent() &&
+                    // TODO: Deprecate compatible table partitioning
+                    !source.getProperties().isCompatibleTablePartitioningWith(partitioningScheme.get().getPartitioning(), false, metadata, session) &&
+                    !(partialMergePushdownStrategy == PUSHDOWN_BUT_NOT_THROUGH_JOIN_AND_AGGREGATION &&
```

Comment:
could this whole thing be captured in canPushDownPartialMerge? Right now here and a few times below you have to separately capture all the logic about when partialmerges get pushed down, which it more confusing and error prone.